gitpod-io/gitpod
# [KOTS]: extract images from the Installer and put in the additionalImages array

## Description

For airgapped mode to work, we need to specify the images used if they are not directly specified in the manifests - the Installer image will be the only one included automatically. As we're using the Installer to manage the container images, we must add these to the `additionalImages` spec.

As we already have the `mirror list` command, this creates a `mirror kots` command to add the `additionalImages` into the `kots-app.yaml` file. This uses the same underlying command as the `mirror list` command, but reworks it so the `original` image is added in - KOTS will then process these images and include them in the internal registry when an airgapped installation is requested.

See [KOTS docs](https://docs.replicated.com/reference/custom-resource-application#additionalimages)

## Related Issue(s)

Fixes #9042

## How to test

Manually trigger an airgap build for [this branch](https://vendor.replicated.com/apps/gitpod/channels/278yCdgOMZvEejkxNTAcnSfQ4UV) and ensure that it complete successfully.

Uploaded `kots-app.yaml` - see [release](https://vendor.replicated.com/apps/gitpod/releases/93):

![image](https://user-images.githubusercontent.com/275848/161018483-a47c15c0-bd04-402e-b97e-4f72205df209.png)

Success message - see [build](https://vendor.replicated.com/apps/gitpod/channels/278yCdgOMZvEejkxNTAcnSfQ4UV):

![image](https://user-images.githubusercontent.com/275848/161025348-4c50f901-c3ea-40f1-927d-875b2019f8c2.png)

## Release Notes

```release-note
[KOTS]: extract images from the Installer and put in the additionalImages array
```

## Documentation
